Sources/ComposableArchitecture/Scheduling/TestScheduler.swift

```diff
@@ -56,20 +56,20 @@ where SchedulerTimeType: Strideable, SchedulerTimeType.Stride: SchedulerTimeInte
     options _: SchedulerOptions?,
     _ action: @escaping () -> Void
   ) -> Cancellable {
-    let id = self.nextId()
+    let sequence = self.nextSequence()
 
     func scheduleAction(for date: SchedulerTimeType) -> () -> Void {
       return { [weak self] in
-        action()
         let nextDate = date.advanced(by: interval)
-        self?.scheduled.append((id, nextDate, scheduleAction(for: nextDate)))
+        self?.scheduled.append((UUID(), sequence, nextDate, scheduleAction(for: nextDate)))
+        action()
```

Comment:
it also wasn't right to execute this interval work immediately and then append the next work onto the queue. We need to do the reverse so that if the work's cancellable is called while invoking `action` then we will clean up _all_ of the interval work and not accidentally leave some stuff behind.